commonmark/src/main/java/org/commonmark/internal/DocumentParser.java

```diff
@@ -63,31 +63,19 @@ public DocumentParser(List<BlockParserFactory> blockParserFactories, InlineParse
     /**
      * The main parsing function. Returns a parsed document AST.
      */
-    public Document parse(String input) {
+    public Document parse(T input) throws IOException
+    {
         DocumentBlockParser documentBlockParser = new DocumentBlockParser();
         documentBlockParser.getBlock().setSourcePosition(new SourcePosition(1, 1));
         activateBlockParser(documentBlockParser);
-
-        int lineStart = 0;
-        int lineBreak;
-        while ((lineBreak = Parsing.findLineBreak(input, lineStart)) != -1) {
-            CharSequence line = Substring.of(input, lineStart, lineBreak);
-            incorporateLine(line);
-            if (lineBreak + 1 < input.length() && input.charAt(lineBreak) == '\r' && input.charAt(lineBreak + 1) == '\n') {
-                lineStart = lineBreak + 2;
-            } else {
-                lineStart = lineBreak + 1;
-            }
-        }
-        if (input.length() > 0 && (lineStart == 0 || lineStart < input.length())) {
-            incorporateLine(Substring.of(input, lineStart, input.length()));
-        }
-
+        
+        incorporateLines(input);
+        
         finalizeBlocks(activeBlockParsers, lineNumber);
         this.processInlines();
         return documentBlockParser.getBlock();
     }
-
+    
```

Comment:
It's not my editor, it's me. I think I do not check for empty line, or just miss it.